src/Admin/AbstractAdmin.php

```diff
@@ -2725,10 +2725,15 @@ public function configureActionButtons($action, $object = null)
             ];
         }
 
-        if (\in_array($action, ['show', 'delete', 'acl', 'history'], true)
-            && $this->hasRoute('edit')
-            && null !== $object
+        $canAccessObject =
+            null !== $object
+            && \in_array($action, ['show', 'edit', 'delete', 'acl', 'history'], true)
```

Comment:
I'm not sure this check improve performances

Reply:
If `$action` is `show` or `edit` then `$this->id($object)` call thrice. As far as i can tell, calling method `AbstractAdmin::id()` is not free. This is a minor performance improvement, but it removes the duplication of code.

I can improve performance for internal actions by removing this line:

```diff
$canAccessObject =
    null !== $object
-    && \in_array($action, ['show', 'edit', 'delete', 'acl', 'history'], true)
    && null !== $this->id($object)
;
```